mlkem/asm/aarch64/rej_uniform_asm_clean.S

```diff
@@ -0,0 +1,644 @@
+    ldp     x29, x30, [sp, #(STACK_OFFSET_GPRS + 32)]
+.endm
+
+    /* Input registers */
+    output                      .req x0
+    len                         .req w1
+    buf                         .req x2
+    buf_consumed_ptr            .req x3
+    buflen                      .req w4
+    table_idx                   .req x5
+    bit_table                   .req x6
+
+    /* Output registers */
+    output_tmp                  .req x11
+    output_tmp_base             .req x16
+
+    /* Number of coefficients sampled so far */
+    count                       .req w6
+    buf_consumed                .req w7
+
+    /* Temporary registers */
+    tmp                         .req w8
+
+    rec_idx_0                   .req w12
+    rec_idx_1                   .req w13
+    rec_idx_2                   .req w14
+    rec_idx_3                   .req w15
+
+    ctr0                        .req w12
+    ctr1                        .req w13
+    ctr2                        .req w14
+    ctr3                        .req w15
+
+    ctr01                       .req ctr0
+    ctr23                       .req ctr2
+
+    /* Vector registers */
+
+    buf0                        .req v0
+    buf1                        .req v1
+    buf2                        .req v2
+
+    tmp0                        .req v4
+    tmp1                        .req v5
+    tmp2                        .req v6
+    tmp3                        .req v7
+
+    sign0                       .req v4
+    sign1                       .req v5
+    sign2                       .req v6
+    sign3                       .req v7
+
+    val0                        .req v16
+    val0q                       .req q16
+    val1                        .req v17
+    val1q                       .req q17
+    val2                        .req v18
+    val2q                       .req q18
+    val3                        .req v19
+    val3q                       .req q19
+
+    t0                          .req s20
+    t1                          .req s21
+    t2                          .req s22
+    t3                          .req s23
+
+    table0                      .req v24
+    table0q                     .req q24
+    table1                      .req v25
+    table1q                     .req q25
+    table2                      .req v26
+    table2q                     .req q26
+    table3                      .req v27
+    table3q                     .req q27
+
+    kyber_q                     .req v30
+    bits                        .req v31
+    bits_q                      .req q31
+
+.text
+.align 4
+.global     rej_uniform_asm_clean
+.global     _rej_uniform_asm_clean
+rej_uniform_asm_clean:
+_rej_uniform_asm_clean:
+    push_stack
+    save_gprs
+
+    // TODO: This routine seems to write the results on the stack first, which is statically sized.
+    //       It therefore only works for input buffers of bounded size, which needs to be documented.
```

Comment:
You can remove this now since you have added the check, right?

Reply:
Yes